cli/tests/integration/lsp_tests.rs

```diff
@@ -1066,6 +1066,136 @@ fn lsp_hover_dependency() {
   );
 }
 
+// Regression test for #12753
+#[test]
+#[ignore]
+fn lsp_hover_keep_type_info_after_invalid_syntax_change() {
```

Comment:
Maybe this is related to #12740?

Reply:
no... because those code paths don't cross at all... I am pretty sure I know what it is, as hinted, when we have a bad parse of a module because of a syntax change, we drop the parsed module, meaning that we can't type check it anymore, therefore we can't get completions for it anymore, therefore the regression.

We will have to retain the parsed module when we get a parse error, so that we can do things like this.